eng/common/TestResources/build-test-resource-config.yml

```diff
@@ -0,0 +1,65 @@
+parameters:
+  - name: SubscriptionConfiguration
+    type: string
+    default: $(sub-config-azure-cloud-test-resources)
+  - name: SubscriptionConfigurations
+    type: object
+    default: null
+
+steps:
+  - ${{ if parameters.SubscriptionConfiguration }}:
+    - pwsh: |
+        $config = @'
+          ${{ parameters.SubscriptionConfiguration }}
+        '@ | ConvertFrom-Json -AsHashtable
+
+        foreach($pair in $config.GetEnumerator()) {
+          if ($pair.Value -is [Hashtable]) {
+            foreach($nestedPair in $pair.Value.GetEnumerator()) {
+              Write-Host "##vso[task.setvariable variable=$($nestedPair.Name);issecret=true;]$($nestedPair.Value)"
+            }
+          } else {
+            Write-Host "##vso[task.setvariable variable=$($pair.Name);issecret=true;]$($pair.Value)"
+          }
+        }
+
+        Write-Host ($config | ConvertTo-Json)
+        $serialized = $config | ConvertTo-Json -Compress
+        Write-Host "##vso[task.setvariable variable=SubscriptionConfiguration;]$serialized"
+      displayName: Initialize SubscriptionConfiguration variable
+
+  - ${{ if parameters.SubscriptionConfigurations }}:
+    - pwsh: |
+        Write-Host "##vso[task.setvariable variable=SubscriptionConfiguration;]{}"
+      displayName: Initialize SubscriptionConfiguration variable
+      condition: eq(variables['SubscriptionConfiguration'], '')
+
+    - ${{ each config in parameters.SubscriptionConfigurations }}:
+      - pwsh: |
+          $config = @'
+            $(SubscriptionConfiguration)
+          '@ | ConvertFrom-Json -AsHashtable
+          $addToConfig = @'
+            ${{ config }}
+          '@ | ConvertFrom-Json -AsHashtable
+
+          foreach ($pair in $addToConfig.GetEnumerator()) {
+            if ($pair.Value -is [Hashtable]) {
```

Comment:
This assumes only one level of nesting is that safe to assume?

Reply:
Technically, someone could update a config they manage to contain more levels of nesting, but it would be ignored by all our scripts anyway, since we explicitly only look for `EnvironmentVariablers` and `ArmTemplateParameters`.

It may be helpful to publish a json schema somewhere for authors of subscription configs. I'm planning on writing a big doc next week to map out all aspects of configuring multi-cloud tests, so I'll include some config schema guidance as well.